client/post-editor/controller.js

```diff
@@ -174,6 +174,14 @@ async function redirectIfBlockEditor( context, next ) {
 	const state = context.store.getState();
 	const siteId = getSelectedSiteId( state );
 
+	// URLs with a set-editor=classic param are used for switching from the block editor to the classic editor, so we
+	// bypass the selected editor check if it is present and update the selected editor for the current user/site pair.
+	const switchToEditor = get( context.query, 'set-editor' );
+	if ( switchToEditor && 'classic' === switchToEditor ) {
+		context.store.dispatch( setSelectedEditor( siteId, 'classic' ) );
```

Comment:
I'm sure it's worth checking the existing editor preference before _setting_ it again. should we do this?

```js
const newEditorChoice = context.query[ 'set-editor' ];
const oldEditorChoice = getSelectedEditor( … );

if ( ! oldEditorChoice || newEditorChoice !== oldEditorChoice ) {
	context.store.dispatch( setSelectedEditor( siteId, newEditorChoice ) );
}

if ( 'classic' === newEditorChoice ) {
	return next();
}
```

written this way I'm separating the logic of updating the editor choice from the logic of redirecting to the classic editor. as the controller I'm wondering if we should be special-casing `classic` or allowing things like `&set-editor=gutenberg` to work as well.

Reply:
Applied some changes in 9ce0249 based on this.